torchgeo/datasets/reforestree.py

```diff
@@ -0,0 +1,301 @@
+
+        image = self._load_image(filepath)
+
+        boxes, labels, agb = self._load_target(filepath)
+
+        sample = {"image": image, "boxes": boxes, "label": labels, "agb": agb}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self, root: str) -> List[str]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, annotation
+        """
+        image_paths = sorted(glob.glob(os.path.join(root, "tiles", "**", "*.png")))
+
+        return image_paths
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array: "np.typing.NDArray[np.int_]" = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, filepath: str) -> Tuple[Tensor, ...]:
+        """Load boxes and labels for a single image.
+
+        Args:
+            filepath: image tile filepath
+
+        Returns:
+            dictionary containing boxes, label, and agb value
+        """
+        tile_df = self.annot_df[self.annot_df["img_path"] == os.path.basename(filepath)]
+
+        boxes: Tensor = torch.Tensor(
+            tile_df[["xmin", "ymin", "xmax", "ymax"]].values.tolist()
+        )
+        labels: Tensor = torch.Tensor(
+            [self.class2idx[label] for label in tile_df["group"].tolist()]
+        )
+        agb: Tensor = torch.Tensor(tile_df["AGB"].tolist())
+
+        return boxes, labels, agb
+
+    def _verify(self) -> None:
+        """Checks the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories are found, else False
+        """
+        filepaths = [os.path.join(self.root, dir) for dir in ["tiles", "mapping"]]
+        if all([os.path.exists(filepath) for filepath in filepaths]):
+            return
+
+        filepath = os.path.join(self.root, self.zipfilename)
+        if os.path.isfile(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory, either specify a different"
```

Comment:
I like to also include what `root` is in the error message, see CDL for an example of this